src/coauthor_interface/backend/PluginInterface.py

```diff
@@ -0,0 +1,29 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+
+
+class InterventionType(str, Enum):
+    # NOTE: this will be updated with new intervention types later. For now, this will only support toasts
+    toast = "toast"
+
+
+class Intervention:
+    def __init__(self, intervention_type: InterventionType, intervention_message: str):
+        self.intervention_dict = {
```

Comment:
This class doesn't feel pythonic. Consider making it a `dataclass` and adding a method that returns a dictionary (I'm assuming that another part of the product will need this dictionary).